src/cli/main.c

```diff
@@ -66,18 +82,44 @@ static void memcpyAndAppendNul(char* dst, char* src, int len) {
     dst[len] = '\0';
 }
 
+// getDictuPath sets up the path to be used for saving history
+// amongst other things. The returned string needs to be freed
+// by the caller.
+char *getDictuPath() {
+    char *dictuPath = calloc(PATH_MAX, sizeof(char) * PATH_MAX);
+    char *tmp;
+
+    if ((tmp = getenv("DICTU_PATH")) != NULL ) {
+        strncpy(dictuPath, tmp, strlen(tmp));
```

Comment:
I'd probably stick with strncat for both operations here as it will null terminate for you, strncpy won't